templates/mariadb/my.cnf.erb

```diff
@@ -18,7 +18,10 @@ gtid_ignore_duplicates  = ON
 gtid_strict_mode        = ON
 log_bin                 = binlog
 log_slave_updates       = ON
-server_id               = <%= @server_id %>
+# Use IP adress as decimal in order to create an unique server id
+server_id               = <%= @facts['networking']['ip'].split(".").map(&:to_i).pack('CCCC').unpack('N')[0] %>
```

Comment:
Are you sure that this can not be done more clearly in the puppet manifest, this is not very readable...